CONTRIBUTING.md

```diff
@@ -0,0 +1,504 @@
+
+- `AWSMobileClient`
+- `AWSCognitoAuth`
+- `AWSS3TransferUtility`
+- `AWSLexInteractionKit`
+- `AWSIoTDataManager`
+- `AWSPinpointTargetingClient`
+- `AWSPinpointAnalyticsClient`
+- `AWSPollySynthesizeSpeechURLBuilder`
+- `AWSTranscribeStreamingService`
+- `AWSMobileAnalyticsEventClient`
+
+You could go into any of these files and start adding new endpoints you see in the generated files to these higher level clients or fixing bugs as this is mostly likely where they would occur since it is not code generated code based off of models given to us by the service teams themselves.
+
+Once you have made your code change, make sure you can build and run both the Tests Folder and the UnitTests folder usually listed below the service folder. The Tests folder will be where the integration tests are and setting up your evironment to run the integration tests is detailed [above](#integration-testing-setup). Unit tests and/or integration tests are required for all features being added to the code-base. Updating existing tests is also required as well. Running tests in this project is the same as any other project in Xcode -> Set the scheme to be the service folder of the service you are editing. Then head to the tests tab on the left hand side of Xcode and run the tests you need to. If I was running Transcribe tests, Xcode would look like this:
+
+![image](readme-images/xcode-tests.png?raw=true)
+
+
+## Debugging
+
+### Running Cocoapods Locally
+
+Framework development is quite different from typical app development when it comes to debugging and being able to test your code. First you will want to create a new app that uses the pods you changed. For instance, if you changed something in the `AWSS3TransferUtility` to test to see if it works, you will need to create a new app and run `pod init` and edit the `Podfile` to use the local version of the SDK you have. This is done by editing the `Podfile` of the new application to take a path to the pod instead of a version number like below:
+
+```ruby
+
+target 'MySampleApp' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+  # with :path I can direct my project to use a local path for the pod
+  pod 'AWSCore', :path => '~/Projects/aws-sdk-ios'      # 
+  pod 'AWSS3TransferUtility', :path => '~/Projects/aws-sdk-ios'           
+  
+end
+```
+Then you want to run `pod install --repo-update` at the root of your sample app to make sure it is using the local version of the pods. When you open the workspace, under the Pods project, the you'll see a section call Development Pods. This where those local pods were installed and if you want to change code here to debug, it will change in the other repo, just make sure you clean and re-build before testing your code again.
```

Comment:
sentence is awkward "the you'll see a section call Development Pods"
and
"This where those local pods"

Reply:
adjusted